### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by iphydf**
This change is [Reviewable](https://reviewable.io/reviews/toktok/toxcore/41)

### CODE REVIEW

**Comment by tux3** [comment]
Review status: 0 of 13 files reviewed at latest revision, 2 unresolved discussions.

---

*[auto_tests/dht_test.c, line 351 \[r1\]](https://reviewable.io:443/reviews/toktok/toxcore/41#-KPTJRIHugTd4YjLNW4H:-KPTJRIIM0dYk5eZpyC6:bsuk7kw) ([raw file](https://github.com/toktok/toxcore/blob/71c0ecdb0a1feee21af5089232b15ce8949756ee/auto_tests/dht_test.c#L351)):*
> ```C
> }
> 
> #if 0
> ```

Why is this code compiled-out?
If the #if 0 is only meant to be temporary, it would be good to have a comment explaining why this should not be removed but should not be run, so readers 10 years from now will not be left wondering :)
If it's entirely broken though, perhaps it belongs in the git history until a fix is available. 

The same remark applies to other commented-out sections below.

---

*[testing/tox_shell.c, line 83 \[r1\]](https://reviewable.io:443/reviews/toktok/toxcore/41#-KPTNQxKQ1NNq00TP5Xt:-KPTMhwx8fORHJtjm3eG:bckosvu) ([raw file](https://github.com/toktok/toxcore/blob/71c0ecdb0a1feee21af5089232b15ce8949756ee/testing/tox_shell.c#L83)):*
> ```C
>     if (ret == -1) {
>         printf("fork failed\n");
>         free(master);
> ```

Are you freeing on return from main to get a clean valgrind output?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/toxcore/41)*

**Comment by tux3** [comment]
Reviewed 13 of 13 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/toxcore/41)*

**Comment by iphydf** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[auto_tests/dht_test.c, line 351 \[r1\]](https://reviewable.io:443/reviews/toktok/toxcore/41#-KPTJRIHugTd4YjLNW4H:-KPTO4g66eF2DCMI85a4:bn3hcnc) ([raw file](https://github.com/toktok/toxcore/blob/71c0ecdb0a1feee21af5089232b15ce8949756ee/auto_tests/dht_test.c#L351)):*
<details><summary><i>Previously, tux3 (Tux3) wrote…</i></summary>
> Why is this code compiled-out?
> If the #if 0 is only meant to be temporary, it would be good to have a comment explaining why this should not be removed but should not be run, so readers 10 years from now will not be left wondering :)
> If it's entirely broken though, perhaps it belongs in the git history until a fix is available. 
> 
> The same remark applies to other commented-out sections below.
</details>
The `DEFTESTCASE` below was commented out already. This change just comments out the tests themselves as well to avoid confusion (i.e. why are these not called?).

---

*[testing/tox_shell.c, line 83 \[r1\]](https://reviewable.io:443/reviews/toktok/toxcore/41#-KPTNQxKQ1NNq00TP5Xt:-KPTOIjkUGlo7DH6wp5_:bakfr00) ([raw file](https://github.com/toktok/toxcore/blob/71c0ecdb0a1feee21af5089232b15ce8949756ee/testing/tox_shell.c#L83)):*
<details><summary><i>Previously, tux3 (Tux3) wrote…</i></summary>
> Are you freeing on return from main to get a clean valgrind output?
</details>
Mostly to make the static analyser happy, but yeah, it'll also make valgrind happier :).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/toxcore/41)*

**Comment by tux3** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[auto_tests/dht_test.c, line 351 \[r1\]](https://reviewable.io:443/reviews/toktok/toxcore/41#-KPTJRIHugTd4YjLNW4H:-KPTP2XQ-3YMX57NNSQz:brm3oi4) ([raw file](https://github.com/toktok/toxcore/blob/71c0ecdb0a1feee21af5089232b15ce8949756ee/auto_tests/dht_test.c#L351)):*
<details><summary><i>Previously, iphydf wrote…</i></summary>
> The `DEFTESTCASE` below was commented out already. This change just comments out the tests themselves as well to avoid confusion (i.e. why are these not called?).
</details>
You're right, they were commented out already. In a spirit of continuous improvement, shouldn't they be removed if no one can explain why they are commented out and not deleted?
Git will do what it does and keep a trace if anyone needs to revive dead code.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/toxcore/41)*

**Comment by iphydf** [comment]
Review status: 12 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

*[auto_tests/dht_test.c, line 351 \[r1\]](https://reviewable.io:443/reviews/toktok/toxcore/41#-KPTJRIHugTd4YjLNW4H:-KPTPwUcU5ZvdOcNmTKC:b39syto) ([raw file](https://github.com/toktok/toxcore/blob/71c0ecdb0a1feee21af5089232b15ce8949756ee/auto_tests/dht_test.c#L351)):*
<details><summary><i>Previously, tux3 (Tux3) wrote…</i></summary>
> You're right, they were commented out already. In a spirit of continuous improvement, shouldn't they be removed if no one can explain why they are commented out and not deleted?
> Git will do what it does and keep a trace if anyone needs to revive dead code.
</details>
I've added a comment and commented out the whole unused block. I'd like to keep the non-working tests around, so we can remember to fix them.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/toxcore/41)*

**Comment by tux3** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/toxcore/41)*

**Comment by tux3** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/toxcore/41#-:-KPTQFeTWpmsq52F1NcR:bnfp4nl)*

File: toxav/rtp.c [hunk 1]
```diff
@@ -382,7 +382,8 @@ int handle_rtp_packet (Messenger *m, uint32_t friendnumber, const uint8_t *data,
                 session->mp = new_message(ntohs(header->tlen) + sizeof(struct RTPHeader), data, length);
 
                 /* Reposition data if necessary */
-                if (ntohs(header->cpart));
+                if (ntohs(header->cpart))
+                    ;
```
Comment (nurupo) [inline toxav/rtp.c]: What is the purpose of this? This statement has no effect on anything.

Comment (GrayHatter) [inline toxav/rtp.c]: this is something that's been in toxav for some time. I had it fixed in a ToxAV branch I was working on, but I abandoned it before it got merged. That's the only reason it still exists, irungentoo and mannol were both aware, but waiting to merge this fix with my AV changes.

**Pull request merged**